### ISSUE TITLE
Anroid: retrieveCalendars is returning an empty list

### ISSUE THREAD

**Issue by dbrpci** [builttoroam/flutter_plugins#560]
**Describe the bug**
On Android 14, retrieveCalendars is returning an empty list for me. I'm seeing this with the example app as well as in our own app that was previously working in the past.

**To Reproduce**
1. Open example app
2. Grant calendar permission
3. See empty screen

**Expected behavior**
It should return a list of calendars on the device

**Screenshots**
n/a

**Device(s) tested**
Pixel 9 Emulator, Android API 35
Pixel 8 Emulator, Android API 34
Pixel 6 device, Android v14

**Flutter doctor**
```
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.22.2, on macOS 14.7 23H124 darwin-arm64, locale
    en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.2)
[✓] VS Code (version 1.93.1)
```

**Additional context**
```
  device_calendar:
    dependency: "direct main"
    description:
      name: device_calendar
      sha256: "683fb93ec302b6a65c0ce57df40ff9dcc2404f59c67a2f8b93e59318c8a0a225"
      url: "https://pub.dev"
    source: hosted
    version: "4.3.3"
    ```

**Comment by thomassth** [comment]
Can't replicate this issue on my end, both before & after I updated my flutter & android SDK.

Maybe it's device / account issue?

**Comment by dbrpci** [comment]
@thomassth - thanks for the quick reply.

I'm using the same google account on multiple devices and emulators. This should work on the emulator, correct?

**Comment by dbrpci** [comment]
FYI - I had a coworker test the example on their machine/device. They saw the same results. No calendars returned on Android, but worked fine for iOS.

I tested [this native android example](https://github.com/hanmajid/modern-android-samples/tree/master/calender-provider) on my same device and it worked fine. So, that gives me some evidence the problem is with this library.

When I get some time, I'm going to setup my debugger with the example and see what's going on, comparing the kotlin code between the two projects.

**Comment by dbrpci** [comment]
Closing this issue.

After a lot of debugging today, I was able to confirm the latest version of the package from the master branch is working on my emulator device. I logged into a google account and could see the list of calendars, events, etc.

@thomassth - thanks for being so responsive, and sorry.

If others run into something similar, double check permissions and also check that you're logged into the google account on the device and maybe reboot the device after logging in.
